### ISSUE TITLE
clusternet-hub took too much memory for lots of failed HelmReleases

### ISSUE THREAD

**Issue by DanielXLee** [clusternet/clusternet#493]
#### What happened:
hub pod started 86m, used memory 8943Mi

```console
➜  ~ k get po
NAME                                    READY   STATUS    RESTARTS   AGE
clusternet-hub-86b5bfc555-zd4rq         1/1     Running   0          86m
clusternet-scheduler-64bf68577d-52kn4   1/1     Running   0          89m
clusternet-scheduler-64bf68577d-q8rp6   1/1     Running   0          89m
clusternet-scheduler-64bf68577d-sjxp5   1/1     Running   0          89m
➜  ~ k top po
NAME                                    CPU(cores)   MEMORY(bytes)
clusternet-hub-86b5bfc555-zd4rq         292m         8943Mi
clusternet-scheduler-64bf68577d-52kn4   1m           13Mi
clusternet-scheduler-64bf68577d-q8rp6   1m           13Mi
clusternet-scheduler-64bf68577d-sjxp5   1m           15Mi
```

#### How to reproduce it (as minimally and precisely as possible):
Use local-running.sh script to create a local kind cluster, create some HelmChart resources, and deployed to 3 child clusters, wait some minutes, and the memory will continue to grow

pprof message:
![image](https://user-images.githubusercontent.com/3355374/192475050-4096f77d-3348-4180-9f8c-28b5ffce0659.png)

use command `go tool pprof -http=:8081 pprof.clusternet-hub.alloc_objects.alloc_space.inuse_objects.inuse_space.005.pb.gz` to see detail
[pprof.clusternet-hub.alloc_objects.alloc_space.inuse_objects.inuse_space.005.pb.gz](https://github.com/clusternet/clusternet/files/9653429/pprof.clusternet-hub.alloc_objects.alloc_space.inuse_objects.inuse_space.005.pb.gz)

#### Environment:
- Clusternet version:
    - Clusternet-agent version (user `clusternet-agent --version=json`):
    - Clusternet-hub version (user `clusternet-hub --version=json`):
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Others:

**Comment by dixudx** [comment]
What's the version of `clusternet-hub`?

**Comment by DanielXLee** [comment]
Base v0.11.0, cherry-pick 4 commit
```bash
* 462c67a - support helm release reinstall (#477) (Thu Sep 15 18:30:30 2022 +0800) <Xin Li>
* cbfc2ea - custom helm install/upgrade extras flags (#467) (Tue Sep 13 10:24:30 2022 +0800) <Xin Li>
* 217814a - use bool point for AppPusher (#474) (Fri Sep 9 17:23:31 2022 +0800) <Xin Li>
* 656f96e - don't override empty original object for jsonpatch (#431) (Sat Jul 23 10:08:18 2022 +0800) <Xin Li>
* e50d81f - (tag: v0.11.0) add use-metrics-server flag in clusternet-agent deployment (#365) (Thu Jul 14 15:21:30 2022 +0800) <Lutzow>
```

**Comment by dixudx** [comment]
```
root@VM-3-30-ubuntu:/home/ubuntu/clusternet# kubectl top pod -n clusternet-system
NAME                                    CPU(cores)   MEMORY(bytes)
clusternet-agent-54fd665874-79nc6       1m           11Mi
clusternet-agent-54fd665874-7p4n7       2m           18Mi
clusternet-agent-54fd665874-pw7bx       1m           11Mi
clusternet-hub-74b6994f64-48lrn         20m          51Mi
clusternet-hub-74b6994f64-5cwvm         24m          59Mi
clusternet-hub-74b6994f64-qfh7t         18m          152Mi
clusternet-scheduler-8485d9f7f7-8mjq7   1m           12Mi
clusternet-scheduler-8485d9f7f7-s9q6r   1m           10Mi
clusternet-scheduler-8485d9f7f7-zxdcj   1m           10Mi
root@VM-3-30-ubuntu:/home/ubuntu/clusternet# kubectl get desc -A
NAMESPACE          NAME               DEPLOYER   STATUS    AGE
clusternet-ftx8w   app-demo-generic   Generic    Success   5m41s
clusternet-ftx8w   app-demo-helm      Helm       Success   5m41s
clusternet-sn9bq   app-demo-generic   Generic    Success   5m41s
clusternet-sn9bq   app-demo-helm      Helm       Success   5m41s
root@VM-3-30-ubuntu:/home/ubuntu/clusternet# kubectl get desc -A
NAMESPACE          NAME               DEPLOYER   STATUS    AGE
clusternet-ftx8w   app-demo-generic   Generic    Success   10m
clusternet-ftx8w   app-demo-helm      Helm       Success   10m
clusternet-sn9bq   app-demo-generic   Generic    Success   10m
clusternet-sn9bq   app-demo-helm      Helm       Success   10m
root@VM-3-30-ubuntu:/home/ubuntu/clusternet# kubectl top pod -n clusternet-system
NAME                                    CPU(cores)   MEMORY(bytes)
clusternet-agent-54fd665874-79nc6       1m           11Mi
clusternet-agent-54fd665874-7p4n7       2m           19Mi
clusternet-agent-54fd665874-pw7bx       1m           11Mi
clusternet-hub-74b6994f64-48lrn         18m          51Mi
clusternet-hub-74b6994f64-5cwvm         14m          59Mi
clusternet-hub-74b6994f64-qfh7t         18m          152Mi
clusternet-scheduler-8485d9f7f7-8mjq7   2m           12Mi
clusternet-scheduler-8485d9f7f7-s9q6r   1m           10Mi
clusternet-scheduler-8485d9f7f7-zxdcj   1m           10Mi
```

Did not notice any abnormal memory usage when using latest code from main branch.

**Comment by DanielXLee** [comment]
Base v0.11.0, cherry-pick commit 656f96e, create 40 hr, failed 30 hr, the memory used about 840Mi
```bash
➜  ~ k get po
NAME                                    READY   STATUS    RESTARTS   AGE
clusternet-hub-864bc9d6cc-nbcnt         1/1     Running   0          123m
clusternet-scheduler-64bf68577d-52kn4   1/1     Running   0          5h9m
clusternet-scheduler-64bf68577d-q8rp6   1/1     Running   0          5h9m
clusternet-scheduler-64bf68577d-sjxp5   1/1     Running   0          5h9m
➜  ~ k top po
NAME                                    CPU(cores)   MEMORY(bytes)
clusternet-hub-864bc9d6cc-nbcnt         7m           840Mi
clusternet-scheduler-64bf68577d-52kn4   1m           13Mi
clusternet-scheduler-64bf68577d-q8rp6   1m           13Mi
clusternet-scheduler-64bf68577d-sjxp5   1m           16Mi
```

Hrs
```bash
➜  ~ k get hr -A
NAMESPACE          NAME                                    CHART             VERSION   REPO                                      STATUS     AGE
clusternet-2ml2b   professional-helm-tke-cert-manager      cert-manager      1.5.5     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-ingress-nginx     ingress-nginx     4.1.4     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-tke-application   tke-application   0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-tke-auth          tke-auth          0.1.2     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-tke-cni           tke-cni           0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   deployed   114m
clusternet-2ml2b   professional-helm-tke-tke-gateway       tke-gateway       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-tke-lb            tke-lb            0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   deployed   114m
clusternet-2ml2b   professional-helm-tke-tke-logagent      tke-logagent      0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-tke-monitor       tke-monitor       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-tke-notify        tke-notify        0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-tke-platform      tke-platform      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-tke-registry      tke-registry      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-2ml2b   professional-helm-tke-tke-storage       tke-storage       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   deployed   114m
clusternet-8r2s6   professional-helm-tke-cert-manager      cert-manager      1.5.5     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-ingress-nginx     ingress-nginx     4.1.4     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-tke-application   tke-application   0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-tke-auth          tke-auth          0.1.2     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-tke-cni           tke-cni           0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   deployed   114m
clusternet-8r2s6   professional-helm-tke-tke-gateway       tke-gateway       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-tke-lb            tke-lb            0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   deployed   114m
clusternet-8r2s6   professional-helm-tke-tke-logagent      tke-logagent      0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-tke-monitor       tke-monitor       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-tke-notify        tke-notify        0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-tke-platform      tke-platform      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-tke-registry      tke-registry      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-8r2s6   professional-helm-tke-tke-storage       tke-storage       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   deployed   114m
clusternet-z5zws   professional-helm-tke-cert-manager      cert-manager      1.5.5     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-ingress-nginx     ingress-nginx     4.1.4     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-tke-application   tke-application   0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-tke-auth          tke-auth          0.1.2     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-tke-cni           tke-cni           0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   deployed   114m
clusternet-z5zws   professional-helm-tke-tke-gateway       tke-gateway       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-tke-lb            tke-lb            0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   deployed   114m
clusternet-z5zws   professional-helm-tke-tke-logagent      tke-logagent      0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-tke-monitor       tke-monitor       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-tke-notify        tke-notify        0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-tke-platform      tke-platform      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-tke-registry      tke-registry      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     114m
clusternet-z5zws   professional-helm-tke-tke-storage       tke-storage       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   deployed   114m
➜  ~ k get hr -A | grep failed
clusternet-2ml2b   professional-helm-tke-cert-manager      cert-manager      1.5.5     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-2ml2b   professional-helm-tke-ingress-nginx     ingress-nginx     4.1.4     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-2ml2b   professional-helm-tke-tke-application   tke-application   0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-2ml2b   professional-helm-tke-tke-auth          tke-auth          0.1.2     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-2ml2b   professional-helm-tke-tke-gateway       tke-gateway       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-2ml2b   professional-helm-tke-tke-logagent      tke-logagent      0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-2ml2b   professional-helm-tke-tke-monitor       tke-monitor       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-2ml2b   professional-helm-tke-tke-notify        tke-notify        0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-2ml2b   professional-helm-tke-tke-platform      tke-platform      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-2ml2b   professional-helm-tke-tke-registry      tke-registry      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-cert-manager      cert-manager      1.5.5     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-ingress-nginx     ingress-nginx     4.1.4     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-tke-application   tke-application   0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-tke-auth          tke-auth          0.1.2     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-tke-gateway       tke-gateway       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-tke-logagent      tke-logagent      0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-tke-monitor       tke-monitor       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-tke-notify        tke-notify        0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-tke-platform      tke-platform      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-8r2s6   professional-helm-tke-tke-registry      tke-registry      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-cert-manager      cert-manager      1.5.5     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-ingress-nginx     ingress-nginx     4.1.4     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-tke-application   tke-application   0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-tke-auth          tke-auth          0.1.2     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-tke-gateway       tke-gateway       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-tke-logagent      tke-logagent      0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-tke-monitor       tke-monitor       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-tke-notify        tke-notify        0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-tke-platform      tke-platform      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
clusternet-z5zws   professional-helm-tke-tke-registry      tke-registry      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed     115m
```

**Comment by dixudx** [comment]
> Base v0.11.0, cherry-pick commit 656f96e, create 40 hr, failed 30 hr, the memory used about 840Mi

That looks reasonable.

> * 462c67a - support helm release reinstall (#477) (Thu Sep 15 18:30:30 2022 +0800) <Xin Li>
> * cbfc2ea - custom helm install/upgrade extras flags (#467) (Tue Sep 13 10:24:30 2022 +0800) <Xin Li>

Now above two commits are quite suspicious for the memory issue.

**Comment by DanielXLee** [comment]
Base v0.11.0, cherry-pick commit 656f96e，217814a，cbfc2ea，462c67a, create 40 hr, failed 30 hr, wait about 13h, the memory used about 14G
```bash
➜  ~ k get po
NAME                                    READY   STATUS    RESTARTS   AGE
clusternet-hub-86b48789bc-9mp7k         1/1     Running   0          13h
clusternet-scheduler-64bf68577d-52kn4   1/1     Running   0          19h
clusternet-scheduler-64bf68577d-q8rp6   1/1     Running   0          19h
clusternet-scheduler-64bf68577d-sjxp5   1/1     Running   0          19h
➜  ~ k top po
NAME                                    CPU(cores)   MEMORY(bytes)
clusternet-hub-86b48789bc-9mp7k         318m         14724Mi
clusternet-scheduler-64bf68577d-52kn4   1m           16Mi
clusternet-scheduler-64bf68577d-q8rp6   1m           15Mi
clusternet-scheduler-64bf68577d-sjxp5   1m           20Mi
```
hrs:
```bash
➜  ~ k get hr -A
NAMESPACE          NAME                                    CHART             VERSION   REPO                                      STATUS            AGE
clusternet-2ml2b   professional-helm-tke-cert-manager      cert-manager      1.5.5     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-ingress-nginx     ingress-nginx     4.1.4     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-2ml2b   professional-helm-tke-tke-application   tke-application   0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-tke-auth          tke-auth          0.1.2     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-tke-cni           tke-cni           0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-tke-gateway       tke-gateway       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-tke-lb            tke-lb            0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   deployed          13h
clusternet-2ml2b   professional-helm-tke-tke-logagent      tke-logagent      0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-tke-monitor       tke-monitor       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-tke-notify        tke-notify        0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-tke-platform      tke-platform      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-tke-registry      tke-registry      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-2ml2b   professional-helm-tke-tke-storage       tke-storage       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-cert-manager      cert-manager      1.5.5     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-ingress-nginx     ingress-nginx     4.1.4     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-8r2s6   professional-helm-tke-tke-application   tke-application   0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-tke-auth          tke-auth          0.1.2     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-tke-cni           tke-cni           0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-tke-gateway       tke-gateway       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-tke-lb            tke-lb            0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   deployed          13h
clusternet-8r2s6   professional-helm-tke-tke-logagent      tke-logagent      0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-tke-monitor       tke-monitor       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-tke-notify        tke-notify        0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-tke-platform      tke-platform      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-tke-registry      tke-registry      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-8r2s6   professional-helm-tke-tke-storage       tke-storage       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-z5zws   professional-helm-tke-cert-manager      cert-manager      1.5.5     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-z5zws   professional-helm-tke-ingress-nginx     ingress-nginx     4.1.4     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-z5zws   professional-helm-tke-tke-application   tke-application   0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-z5zws   professional-helm-tke-tke-auth          tke-auth          0.1.2     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-z5zws   professional-helm-tke-tke-cni           tke-cni           0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-z5zws   professional-helm-tke-tke-gateway       tke-gateway       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-z5zws   professional-helm-tke-tke-lb            tke-lb            0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   deployed          13h
clusternet-z5zws   professional-helm-tke-tke-logagent      tke-logagent      0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-z5zws   professional-helm-tke-tke-monitor       tke-monitor       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-z5zws   professional-helm-tke-tke-notify        tke-notify        0.1.0     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
clusternet-z5zws   professional-helm-tke-tke-platform      tke-platform      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-z5zws   professional-helm-tke-tke-registry      tke-registry      0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   pending-install   13h
clusternet-z5zws   professional-helm-tke-tke-storage       tke-storage       0.1.1     oci://ccr.ccs.tencentyun.com/danielxxli   failed            13h
```

**Comment by dixudx** [comment]
> Base v0.11.0, cherry-pick commit 656f96e, create 40 hr, failed 30 hr, the memory used about 840Mi

> Base v0.11.0, cherry-pick commit 656f96e，217814a，cbfc2ea，462c67a, create 40 hr, failed 30 hr, wait about 13h, the memory used about 14G

Two commits cbfc2ea and 462c67a are quite suspicious for the memory issue. Some extra options were introduced in these two commits, including `wait`, `timeOut` and `xxxHooks`. These settings did have some apparent affects on the memory usage.

**Comment by DanielXLee** [comment]
The root cause is the helm SDK memory leak, fixed in 3.8.1, we can upgrade helm to the latest version
https://github.com/helm/helm/pull/10486
